src/main/java/co/rsk/bitcoinj/script/RedeemScriptUtils.java

```diff
@@ -0,0 +1,40 @@
+package co.rsk.bitcoinj.script;
+
+import co.rsk.bitcoinj.core.ScriptException;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RedeemScriptUtils {
+    private static final Logger logger = LoggerFactory.getLogger(RedeemScriptUtils.class);
+
+    private RedeemScriptUtils() { }
+    public static Optional<RedeemScriptParser> extractRedeemScriptParserFromInputScript(Script inputScript) {
```

Comment:
Not sure about this, it seems to me that the method is doing 2 things. Extracting the redeem script and also parsing it. Usually we want methods to do 1 thing.

I would only extract the redeem script from the input here, then the caller can decide to parse it

Reply:
okidoki. on it